### PR TITLE
Basic handling for consumer cancel

### PR DESCRIPTION
**Pull request by aseppala**
Fixes an issue when RabbitMQ server sends *basic.cancel* and client stops consuming but doesn't kill logstash process. In this case the process goes into a limbo state when no messages are being consumed, but process stays alive.

This patch solves this issue by logging a message when *basic.cancel* is received and stops the process.

Related to #51

### CODE REVIEW

**Comment by andrewvc** [comment]
Apologies @aseppala , someone this missed our review. Would you mind signing the CLA and rebasing this patch? I'd love to include it in our next release if possible.

**Comment by aseppala** [comment]
Rebased and signed CLA. How often does the CLA check run?

File: lib/logstash/inputs/rabbitmq.rb [hunk 1]
```diff
@@ -135,6 +136,10 @@ def shutdown_consumer
         @consumer.gracefully_shut_down
       end
 
+      def on_cancellation
+        @logger.info("Reveived basic.cancel from #{rabbitmq_settings[:host]}, shutting down.")
```
Comment (andrewvc) [inline lib/logstash/inputs/rabbitmq.rb]: s/Reveived/Revived/

**Comment by andrewvc** [comment]
@aseppala there's a slight typo in your PR, I just added a comment. It still is telling me the CLA is not signed, I can check with the CLA guys to see if that's broken... will update you.

**Comment by aseppala** [comment]
It might also be that my company CLA is not signed. I can check also that...

**Comment by andrewvc** [comment]
@aseppala I've pinged the developer behind our CLA check to investigate it. Once we figure that out and you push a correction for the spelling error I'm glad to merge this in. Thank you so much for the excellent patch!

File: lib/logstash/inputs/rabbitmq.rb (hunk 1 above)
Comment (aseppala) [inline lib/logstash/inputs/rabbitmq.rb]: fixed: s/Reveived/Received/

**Comment by karmi** [comment]
Hi @aseppala, I've looked at the issue and re-run the CLA checks, all is green now.

**Comment by stefanmeisner** [comment]
I'm not sure if this is the right place to put this comment but....here it comes anyway ;-)
We are running RabbitMQ in a cluster and logstash is reaching it through a loadbalancer. For me, the only reasonable reaction from logstash is to reconnect to RabbitMQ if it gets a basic.cancel. Logstash is the only application we need to occasionally restart when a RabbitMQ nodes has been shut down

**Comment by michaelklishin** [comment]
@stefanmeisner `basic.cancel` is not sent when a node is restarted. It is sent when a queue is deleted or queue master node goes down and for some reason RabbitMQ cannot automatically re-register the consumer. Consumer cancellation is not covered by automatic connection recovery. If cancelled consumers were to automatically recover, there would be no way to delete a queue that has consumers (it would immediately be redeclared). That sounds like a good idea until it's not.

**Pull request closed without merging**